subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/GradleKotlinDslIntegrationTest.kt

```diff
@@ -293,6 +294,25 @@ class GradleKotlinDslIntegrationTest : AbstractPluginIntegrationTest() {
         )
     }
 
+    @Test
+    @Issue("https://github.com/gradle/gradle/issues/24503")
+    fun `can use configuration provider in dependencies block`() {
+
+        withClassJar("fixture.jar", DeepThought::class.java)
+
+        withBuildScript(
+            """
+            val foo by configurations.registering
+
+            dependencies {
+              foo(files("fixture.jar"))
+            }
+            """
+        )
+
+        build("dependencies")
```

Comment:
Shouldn't this test actually compute something based on `DeepThought` and check the result, to make sure the class is available?

Reply:
Okay, I improved the test logic.